### ISSUE TITLE
Use puremagic module to find obfuscated extesion

### ISSUE THREAD

**Issue by Safihre** [sabnzbd/sabnzbd#1913]
Some downloads contain fully obfuscated files in such a way that even the extensions are obfuscated.
Proposal: use the [`puremagic`](https://github.com/cdgriffith/puremagic/) module to find the most likely extension of an obfuscated file.
We do need to determine when we will run the extension-detection, how do we detect a nonsense extension from a regular one?
It would then also only be activated if Deobfuscate final filenames is turned _on_.

@sanderjo has already been testing this module a bit.

**Comment by sanderjo** [comment]
Example:

In my download:

`7986c213700bb0.8edb8bca0a2fcf7d1237e2cdf082f6.44c8`

the Linux command `file` (which puremagic can probably do too) tells

`PNG image data, 1920 x 1080, 8-bit/color RGB, non-interlaced`

So useful for user to let SAB rename that file to `7986c213700bb0.8edb8bca0a2fcf7d1237e2cdf082f6.44c8.png`  (so with correct extension)

... and after that, deobfuscate() could rename to jobname


@Safihre  is that what you mean?

**Comment by Safihre** [comment]
Yes indeed.

**Comment by puzzledsab** [comment]
You should make sure that the user gets a warning if they are included in the unwanted extension list or not in the white list, and that they are deleted if they are listed in the cleanup list.

**Comment by sanderjo** [comment]
I propose this:

```
def correct_file_extension(file):

    return extension_needs_correction, proposed_file
```

- correct_file_extension("mypicture.png") => False, None
- correct_file_extension("blabla") => True, "blabla.png"

or

```
def correct_file_extension(file):

    return proposed_file # None if no improved extension needed or possible
```

- correct_file_extension("mypicture.png") => None
- correct_file_extension("blabla") => "blabla.png"

Feedback?

**Comment by sanderjo** [comment]
Thanks for your feedback.

> You should make sure that it works with the unwanted extension list 

Unwanted extension is during the download. Deobfuscation is later on. So there is a conflict. Oh ... that is already the case: deobfuscation's par2-based renaming can already now result in unwanted extensions. 
Ah ... nice way for bad guys to get their "clickthis.exe" onto someone's harddrive anyway ... 

> and that they are deleted if they are listed in the cleanup list.

I think cleanup is at the very end, so no problem.

**Comment by puzzledsab** [comment]
How about adding a test in renamer that gives a warning if the new name has an extension that is not wanted?  Should be simple and it works for both cases of deobfuscation. It's too late to pause it anyway.

**Comment by Safihre** [comment]
Closed by #1914
